v03_pipeline/lib/misc/allele_registry.py

```diff
@@ -0,0 +1,215 @@
+    base_url: str
+    error_type: str
+    description: str
+    message: str
+    input_line: str | None
+
+    @classmethod
+    def from_api_response(cls, response: dict, base_url: str):
+        return cls(
+            base_url=base_url,
+            error_type=response['errorType'],
+            description=response['description'],
+            message=response['message'],
+            input_line=response.get('inputLine'),
+        )
+
+    def __str__(self) -> str:
+        msg = (
+            f'\nAPI URL: {self.base_url}\nTYPE: {self.error_type}'
+            f'\nDESCRIPTION: {self.description}\nMESSAGE: {self.message}'
+        )
+        return (
+            msg if self.input_line is None else f'{msg}\nINPUT_LINE: {self.input_line}'
+        )
+
+
+def register_alleles_in_chunks(
+    ht: hl.Table,
+    reference_genome: ReferenceGenome,
+    base_url: str = ALLELE_REGISTRY_URL,
+    chunk_size: int = MAX_VARIANTS_PER_REQUEST,
+):
+    num_rows = ht.count()
+    num_chunks = math.ceil(num_rows / chunk_size)
+    logger.info(
+        f'Registering {num_rows} allele(s) in chunks of {chunk_size} in {num_chunks} request(s).',
+    )
+    for start_idx in range(0, num_rows, chunk_size):
+        end_idx = start_idx + chunk_size
+        if end_idx == chunk_size:
+            chunk_ht = ht.head(chunk_size)
+        elif end_idx <= num_rows:
+            chunk_ht = ht.head(end_idx).tail(chunk_size)
+        else:
+            chunk_ht = ht.tail(end_idx - num_rows)
+        yield register_alleles(chunk_ht, reference_genome, base_url)
+
+
+def register_alleles(
+    ht: hl.Table,
+    reference_genome: ReferenceGenome,
+    base_url: str,
+) -> hl.Table:
+    uuid4 = uuid.uuid4()
+    raw_vcf_file_name = f'{Env.HAIL_TMPDIR}/r_{uuid4}.vcf'
+    formatted_vcf_file_name = f'{Env.HAIL_TMPDIR}/f_{uuid4}.vcf'
+
+    # Export the variants to a VCF
+    hl.export_vcf(ht, raw_vcf_file_name)
+
+    # Reformat the VCF created by hail's 'export_vcf' function to be compatible with the Allele Registry
+    with hfs.open(raw_vcf_file_name, 'r') as vcf_in, hfs.open(
+        formatted_vcf_file_name,
+        'w',
+    ) as vcf_out:
+        vcf_out.writelines(reference_genome.allele_registry_vcf_header)
+        for line in vcf_in:
+            if not line.startswith('#'):
+                # NB: The Allele Registry does not accept contigs prefixed with 'chr', even for GRCh38
+                vcf_out.write(line.replace('chr', ''))
+
+    logger.info('Calling the ClinGen Allele Registry')
+    with hfs.open(formatted_vcf_file_name, 'r') as vcf_in:
+        data = vcf_in.read()
+        res = requests.put(
+            url=build_url(base_url, reference_genome),
+            data=data,
+            timeout=HTTP_REQUEST_TIMEOUT_S,
+        )
+        return handle_api_response(res, base_url, reference_genome)
+
+
+def build_url(base_url: str, reference_genome: ReferenceGenome) -> str:
+    login, password = _get_ar_credentials_from_secret_manager()
```

Comment:
have we tested this yet?  do we need to do something like [this stack overflow suggests](https://stackoverflow.com/a/75878048)?

Reply:
good catch. I tested it in my julia cluster and got "Request is missing required authentication credential."
let me see if I can create a cluster with the cloud-platform scope 

in the meantime I added the scope here: https://github.com/broadinstitute/seqr-pipeline-airflow/pull/241/files#diff-e02c8035fb33918e7057de3b717c24d2f123f31e383b6558acc2808bb24851c9R168